x-pack/plugins/infra/server/lib/metrics/make_get_metric_indices.test.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { savedObjectsClientMock } from '@kbn/core/server/mocks';
+import { defaultSourceConfiguration, InfraSource } from '../sources';
+import { createInfraSourcesMock } from '../sources/mocks';
+import { makeGetMetricIndices } from './make_get_metric_indices';
+
+describe('getMetricIndices', () => {
+  it('should return the indices from a resolved configuration', async () => {
+    const sourceConfiguration: InfraSource = {
+      id: 'default',
+      origin: 'stored',
+      configuration: defaultSourceConfiguration,
+    };
+    const infraSourcesMock = createInfraSourcesMock();
+    infraSourcesMock.getSourceConfiguration.mockReturnValueOnce(
+      Promise.resolve(sourceConfiguration)
+    );
```

Comment:
this works :+1: there's also a jest shorthand for promises [`mockFn.mockResolvedValueOnce()`](https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvalueoncevalue)

Reply:
Nice!